src/System.CommandLine/Invocation/InvocationContext.cs

```diff
@@ -66,6 +69,27 @@ public CancellationToken GetCancellationToken()
             return _cts.Token;
         }
 
+        /// <summary>
+        /// Set <see cref="InvokeResult"/> to the result of the command that was invoked.
+        /// </summary>
+        /// <param name="value">The result of the command invocation. When a task, it's assumed to already have been awaited.</param>
+        internal void SetInvokeResult(object value)
+        {
+            if (value is Task task)
+            {
+                var result = task.GetType().GetProperty("Result").GetValue(value);
```

Comment:
Just a warning, calling `Task<T>.Result` should generally be avoided. If the task is not yet completed this can cause a deadlock under certain circumstances. https://docs.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming

Anyway, I think if we move this evaluation into the `InvocationPipeline` you won't need to do anything like this.

Reply:
Would you like me to try make some of these changes?